### PR TITLE
loosen version constraints on test-kitchen

### PR DESCRIPTION
**Pull request by jamesdehart**
Current version is 2.3.3 and the transport still works.

I also removed `version.rb` file and hardcoded the version as a string 
in
gemspec to avoid loading issues.

`Kitchen::Transport::Rsync` is a class, but the version file defined it
as a module and kitchen would fail loading the plugin:

```
TypeError: Rsync is not a class
  
/home/rob/projects/kitchen-transport-rsync/lib/kitchen/transport/rsync.rb:23:in 
`<module:Transport>'
  
/home/rob/projects/kitchen-transport-rsync/lib/kitchen/transport/rsync.rb:22:in 
`<module:Kitchen>'
  
/home/rob/projects/kitchen-transport-rsync/lib/kitchen/transport/rsync.rb:21:in 
`<top (required)>'
```

There are no references to `VERSION` constant outside of gemspec file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [![Gem Version](https://badge.fury.io/rb/kitchen-transport-rsync.svg)](http://badge.fury.io/rb/kitchen-transport-rsync)
 
 This transport is based on rsync over ssh providing dramatic performance improvements.
-It will only work with TK 1.4 and rsync has to be available in the PATH on both local and remote end points.
+It will only work with if rsync is be available in the PATH on both local and remote end points.
```
Comment (tcondit) [inline README.md:7]: Please change to "It will only work if rsync is available ...".

File: kitchen-transport-rsync.gemspec [hunk 2]
```diff
@@ -33,7 +31,7 @@ Gem::Specification.new do |spec|
   spec.test_files = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = %w(lib)
 
-  spec.add_dependency 'test-kitchen', '~> 1.4'
+  spec.add_dependency 'test-kitchen', '~> 2.3.3'
```
Comment (tcondit) [inline kitchen-transport-rsync.gemspec:34]: Does this mean versions thru 2.3.3 are allowed, and no others?

File: README.md (hunk 1 above)
Comment (jamesdehart) [reply README.md:7]: This is the original source.

File: kitchen-transport-rsync.gemspec (hunk 2 above)
Comment (jamesdehart) [reply kitchen-transport-rsync.gemspec:34]: Less then or equal to `2.3.3`

**Comment by jamesdehart** [comment]
@tcondit Thanks for the comments. I created an issue in regards to the readme update.

File: kitchen-transport-rsync.gemspec (hunk 2 above)
Comment (tcondit) [reply kitchen-transport-rsync.gemspec:34]: Same

**Comment by arthurlogilab** [comment]
:tada:

**Pull request merged**